frontend/src/pages/Results.js

```diff
@@ -0,0 +1,164 @@
+import { useState, useEffect } from 'react'
+import styled from 'styled-components/macro'
+import getFromApi from '../lib/getFromApi'
+import PropTypes from 'prop-types'
+import Map from '../components/Map'
+import controlsSrc from '../assets/controls.svg'
+import FilterMenu from '../components/FilterMenu'
+import TrackCard from '../components/TrackCard'
+import Footer from '../components/Footer'
+import saveLastPositionLocally from '../lib/saveLastPositionLocally'
+import getLastSavedPosition from '../lib/getLastSavedPosition'
+
+Results.propTypes = {
+  startingPoint: PropTypes.object.isRequired,
+}
+
+export default function Results({ startingPoint }) {
+  const lastSearchedPosition = getLastSavedPosition()
+  const [filterCriteria, setFilterCriteria] = useState({
+    distance: 300000,
+  })
+  const [tracks, setTracks] = useState([])
+
+  const [centerCoords, setCenterCoords] = useState({
+    lat: startingPoint.latitude || lastSearchedPosition.lat,
+    lng: startingPoint.longitude || lastSearchedPosition.lng,
+  })
+  const [isFilterActive, setIsFilterActive] = useState(false)
+
+  useEffect(() => {
+    saveLastPositionLocally(centerCoords)
+    const path = `track/${centerCoords.lat},${centerCoords.lng}`
+    getFromApi(path)
+      .then(({ data }) => setTracks(data))
+      .catch((error) => console.error('Error:', error))
+  }, [centerCoords])
+
+  const filteredTracks = tracks.filter((track) => {
+    for (const key in filterCriteria) {
+      if (track[key] === undefined) {
+        return false
+      } else if (
+        (key === 'distance' || key === 'lengthM') &&
+        track[key] > filterCriteria[key]
+      ) {
+        return false
+      } else if (key === 'roundtrip' && track[key] !== filterCriteria[key]) {
+        return false
+      } else if (key === 'difficulty' && track[key] !== filterCriteria[key]) {
+        return false
+      } else if (
+        key === 'certYear' &&
+        Boolean(track[key]) !== filterCriteria[key]
+      ) {
+        return false
+      }
+    }
+    return true
+  })
```

Comment:
If it really has to be in FE, you should refactor

Reply:
In a later branch more filter logic should happen in the backend. Thanks for this impulse.